airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -494,9 +494,114 @@ def validate_start_time(self, requested_start_time: int, value: int = 1) -> int:
         return super().validate_start_time(requested_start_time, value=3)
 
 
-class TicketSubstream(HttpSubStream, IncrementalZendeskSupportStream):
+class TicketMetrics(SourceZendeskSupportStream):
 
-    cursor_field = "generated_timestamp"
+    name = "ticket_metrics"
+    cursor_field = "_ab_updated_at"
+
+    def __init__(self, subdomain: str, start_date: str, ignore_pagination: bool = False, **kwargs):
+        super().__init__(subdomain=subdomain, start_date=start_date, ignore_pagination=ignore_pagination, **kwargs)
+        self._state_migrator = TicketMetricsStateMigration()
+        self._parent_stream: SourceZendeskSupportStream = None
+        self._stateless_ticket_metrics = StatelessTicketMetrics(
+            subdomain=subdomain, start_date=start_date, ignore_pagination=ignore_pagination, **kwargs
+        )
+        self._stateful_ticket_metrics = StatefulTicketMetrics(
+            parent=Tickets(subdomain=subdomain, start_date=start_date, ignore_pagination=ignore_pagination, **kwargs),
+            subdomain=subdomain,
+            start_date=start_date,
+            ignore_pagination=ignore_pagination,
+            **kwargs,
+        )
+
+    @property
+    def parent_stream(self):
+        return self._parent_stream
+
+    @parent_stream.setter
+    def parent_stream(self, stream):
+        self._parent_stream = stream
+
+    def _get_parent_stream(self, stream_state: Mapping[str, Any]) -> SourceZendeskSupportStream:
+        return self._stateful_ticket_metrics if stream_state else self._stateless_ticket_metrics
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        return self.parent_stream.get_updated_state(current_stream_state=current_stream_state, latest_record=latest_record)
+
+    def stream_slices(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        self.parent_stream = self._get_parent_stream(stream_state)
+        yield from self.parent_stream.stream_slices(sync_mode=sync_mode, cursor_field=cursor_field, stream_state=stream_state)
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: Optional[List[str]] = None,
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        stream_state: Optional[Mapping[str, Any]] = None,
+    ) -> Iterable[StreamData]:
+        stream_state = self._state_migrator.migrate(stream_state)
```

Comment:
why do we call migrate everytime we call `read_records()`? assuming there isn't an edge case I'm missing, what if we were to call it during `stream_slices()` so it only needs to be invoked once instead of per-slice.